src/Dictionaries/RangeHashedDictionary.cpp

```diff
@@ -413,7 +421,7 @@ void RangeHashedDictionary::getItemsImpl(
 
             if (val_it != std::end(ranges_and_values))
             {
-                auto& value = val_it->value;
+                auto & value = val_it->value;
```

Comment:
Keys found should be increased here. First we find ranges for key with ```const auto it = attr.find(ids[row]);```. If iterator is valid that means that we can check some ranges for requested key. If ```if (val_it != std::end(ranges_and_values))``` that means that key exists in range.